### PR TITLE
DAR-1

### PR DESCRIPTION
**Pull request by katsimonS**
add controllers

### CODE REVIEW

File: src/Darnytsia.Creatio.Api/Controllers/BooksController.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private readonly IMediator _mediator;
+
+    public BooksController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [Route("{bookId}")]
+    public async Task<IHttpActionResult> GetBookById([FromUri] Guid bookId)
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Api/Controllers/BooksController.cs]:
```suggestion
    public async Task<IHttpActionResult> GetBookByIdAsync([FromUri] Guid bookId)
```

File: src/Darnytsia.Creatio.Api/Controllers/BooksController.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        return Ok(await _mediator.Send(new GetBookByIdQuery(bookId)));
+    }
+
+    [HttpPost]
+    [Route("{bookId}/update")]
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Api/Controllers/BooksController.cs]:
```suggestion
    [Route("{bookId}/filling-authors")]
```

File: src/Darnytsia.Creatio.Api/Controllers/BooksController.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    [SwaggerResponse(HttpStatusCode.Created)]
+    public async Task<IHttpActionResult> Update([FromUri]Guid bookId)
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Api/Controllers/BooksController.cs]:
```suggestion
    public async Task<IHttpActionResult> FillingAuthorsAsync([FromUri]Guid bookId)
```

File: src/Darnytsia.Creatio.Api/Controllers/ContactsController.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    public ContactsController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [Route("{contactId}")]
+    [SwaggerResponse(HttpStatusCode.OK)]
+    public async Task<IHttpActionResult> GetById([FromUri] Guid contactId)
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Api/Controllers/ContactsController.cs]:
```suggestion
    public async Task<IHttpActionResult> GetByIdAsync([FromUri] Guid contactId)
```

File: src/Darnytsia.Creatio.Api/Controllers/ContactsController.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        return Ok(await _mediator.Send(new GetContactBirthDayByIdQuery(contactId)));
+    }
+
+    [HttpPost]
+    [Route("{contactId}/update-birthday")]
+    [SwaggerResponse(HttpStatusCode.Created, Type = typeof(ContactRequest))]
+    public async Task<IHttpActionResult> Update([FromUri]Guid contactId, [FromBody]ContactRequest contact)
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Api/Controllers/ContactsController.cs]:
```suggestion
    public async Task<IHttpActionResult> UpdateBirthdayAsync([FromUri]Guid contactId, [FromBody]ContactRequest contact)
```

File: src/Darnytsia.Creatio.Core/Features/Books/Commands/UpdateBookAuthorCommand.cs [hunk 6]
```diff
@@ -0,0 +1,5 @@
+﻿using System;
+
+namespace Darnytsia.Creatio.Core.Features.Books.Commands;
+
+public record UpdateBookAuthorCommand(Guid BookId) : IRequest;
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Core/Features/Books/Commands/UpdateBookAuthorCommand.cs]: should rename to FillingBookAuthorsCommand

File: src/Darnytsia.Creatio.Core/Features/Books/Handlers/UpdateBookAuthorHandler.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+﻿using Darnytsia.Creatio.Abstractions;
+using Darnytsia.Creatio.Core.Features.Books.Commands;
+using Darnytsia.Creatio.Data.Entities;
+using Darnytsia.Creatio.Data;
+using System.Linq;
+using System.Threading;
+
+namespace Darnytsia.Creatio.Core.Features.Books.Handlers;
+
+public class UpdateBookAuthorHandler : IRequestHandler<UpdateBookAuthorCommand>
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Core/Features/Books/Handlers/UpdateBookAuthorHandler.cs]: should be renamed to FillingBookAuthorsHandler
command, query, handler may not always contain insert, get, select, update, delete. Name can be in any understandable form

File: src/Darnytsia.Creatio.Core/Features/Contacts/Commands/UpdateContactBirthDayCommand.cs [hunk 8]
```diff
@@ -0,0 +1,5 @@
+﻿using System;
+
+namespace Darnytsia.Creatio.Core.Features.Contacts.Commands;
+
+public record UpdateContactBirthDayCommand(Guid ContactId, DateTime BDay) : IRequest;
```
Comment (DyshliukRostyslav) [inline src/Darnytsia.Creatio.Core/Features/Contacts/Commands/UpdateContactBirthDayCommand.cs]:
```suggestion
public record UpdateContactBirthDayCommand(Guid ContactId, DateTime Birthday) : IRequest;
```
try not to use short form of words

**Review by DyshliukRostyslav — CHANGES_REQUESTED** [review]

File: src/Darnytsia.Creatio.Core/Features/Books/Commands/UpdateBookAuthorCommand.cs (hunk 6 above)
Comment (katsimonS) [reply src/Darnytsia.Creatio.Core/Features/Books/Commands/UpdateBookAuthorCommand.cs]: changed

File: src/Darnytsia.Creatio.Core/Features/Books/Handlers/UpdateBookAuthorHandler.cs (hunk 7 above)
Comment (katsimonS) [reply src/Darnytsia.Creatio.Core/Features/Books/Handlers/UpdateBookAuthorHandler.cs]: changed

**Review by DyshliukRostyslav — APPROVED** [review]

**Pull request merged**